packages/react/src/SegmentedControl/SegmentedControl.tsx

```diff
@@ -210,9 +182,12 @@ const Root: React.FC<React.PropsWithChildren<SegmentedControlProps>> = ({
                 isUncontrolled && setSelectedIndexInternalState(index)
               },
           selected: index === selectedIndex,
-          sx: {
+          style: {
             '--separator-color':
               index === selectedIndex || index === selectedIndex - 1 ? 'transparent' : theme?.colors.border.default,
+            ...child.props.style,
+          },
+          sx: {
```

Comment:
```suggestion
          sx: child.props.sx
```

Maybe we just need this now if we're just spreading it into an empty object